File: backend/app/api/routes/applications.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.api.deps import get_current_user
from app.db.session import get_db
from app.models.user import User
from app.schemas.application import ApplicationResponse
from app.services.applications import service as application_service

router = APIRouter(prefix="/applications", tags=["applications"])


@router.get("", response_model=list[ApplicationResponse])
def list_applications(
    db: Session = Depends(get_db),
    _current_user: User = Depends(get_current_user),
) -> list[ApplicationResponse]:
    """The application catalog — visible to any authenticated user, same
    reasoning as the employee directory (`GET /employees`): you need to see
    what's requestable before you can request it. Write access isn't a V1
    concern; see ApplicationResponse's docstring."""
    return application_service.list_applications(db)

File: frontend/src/api/applications.ts
import { API_BASE_URL } from './client'

export interface ApplicationResponse {
  id: string
  name: string
  description: string
  risk_level: string
  created_at: string
}

interface ApiErrorBody {
  error?: { type: string; message: string }
}

/**
 * GET /applications is visible to any authenticated user — you need to see
 * what's requestable before you can request it. No create/update endpoint
 * exists (the catalog is seed-managed, see app/schemas/application.py's
 * docstring), so there's no corresponding write call here.
 */
export async function fetchApplications(token: string): Promise<ApplicationResponse[]> {
  const response = await fetch(`${API_BASE_URL}/applications`, {
    headers: { Authorization: `Bearer ${token}` },
  })
  if (!response.ok) {
    const body = (await response.json().catch(() => null)) as ApiErrorBody | null
    throw new Error(body?.error?.message ?? 'Failed to load applications')
  }
  return response.json() as Promise<ApplicationResponse[]>
}
